### ISSUE TITLE
Some vector layers on more distant tiles are not rendered for pitched 3D terrain map

### ISSUE THREAD

**Issue by zdila** [maplibre/maplibre-gl-js#1618]
**maplibre-gl-js version**: 2.4.0

**browser**: Vivaldi, Firefox

### Steps to Trigger Behavior

 1. create a map with vector style and 3D terrain
 2. zoom it, pitch it

### Link to Demonstration

https://labs.maptiler.com/showcase/osm-3d-terrain/#map=13.05/48.8144/21.5301/59.28/75.50&language=sk

### Expected Behavior

All visible layers should be rendered.

### Actual Behavior

Some vector layers on more distant tiles are not rendered for pitched 3D terrain map.

![image](https://user-images.githubusercontent.com/636095/188572709-75737f0d-3ecc-4005-b61c-ac0c66a24b56.png)

**Comment by HarelM** [comment]
This is by design.
To improve performance and make sure we are not loading too many tiles, the distant tiles are using a lower zoom level tiles.
In your case, since the pitch is over 60 degrees and the horizon is visible, if this algorithm was not implemented the entire world tiles would have needed to be downloaded and rendered. Not a great solution :-)
I guess you'll be able to see more details when you pan the map further into the distance, right?

**Comment by zdila** [comment]
@HarelM that is not the case. You can see gray residential areas and streets but not forests. If you disable 3D (or set pitch to zero) and will zoom out, you will see that residential areas and residential and other minor roads disappear before forests. There is no zoom with tiles containing those roads without forests in that style.

**Comment by HarelM** [comment]
I see. Thanks for the info!
Can you try and create a simplified style that shows this issue maybe?

**Comment by zdila** [comment]
I will do it.

**Comment by zdila** [comment]
Here is the minimal-style demo: https://codesandbox.io/s/maplibre-gl-js-issue-1618-8wzxue

Too see something you must zoom in, for example: https://8wzxue.csb.app/#map=14.28/48.85968/19.36918/-101.31/69.90

There are two layers - one shows landcover to zoom 13 and other shows landcover from zoom 13. If there is only one landcover layer used for all zooms, then the issue is not reproducible.

**Comment by HarelM** [comment]
I only see one layer in the above link... am I missing something?
Not that the issue does not exists there, but just want to make sure I'm looking at the right example...
<img width="973" alt="image" src="https://user-images.githubusercontent.com/3269297/188906570-92bfdf99-f31f-433c-9bd5-4e1bfed33c7a.png">

**Comment by zdila** [comment]
In `style.json` there are two layers. I am actually not counting the third one - shading.

**Comment by prozessor13** [comment]
Hi, this is related to LOD (level of detail). You defined a `minZoom` of 13, tiles near the horizon are requested for 12.

**Comment by zdila** [comment]
There are two layers - one has `"minzoom": 13` other has `"maxzoom": 13`. Affected tiles have nothing there from neither of the layers.

**Comment by zdila** [comment]
@HarelM no problem. I've updated the testcase. Let me know if I could do anything more.

**Comment by prozessor13** [comment]
Tested by myself, and as expected, this issue is also in 2D. So this is no Terrain-3D issue. It is an LOD issue when tilting map more that 60deg.

**Comment by HarelM** [comment]
I don't see the problem when terrain is "off", @prozessor13 can you share what you saw?
Terrain commented out:
![image](https://user-images.githubusercontent.com/3269297/189088189-ad00192e-1db7-40fa-abc8-6c8b99976841.png)
Terrain not commented out:
![image](https://user-images.githubusercontent.com/3269297/189088282-11eec529-6992-4958-b580-1364048fe29f.png)

**Comment by zdila** [comment]
> is this issue also in 2d

"Partially" :-)

If 3D terrain is not enabled, then I see also more distant tiles, but not "to the infinity". I would expect to see the horizont in the red colour (as for zoom < 13 the styling is red).

Samples:

1. 3D enabled:

![image](https://user-images.githubusercontent.com/636095/189095339-49958946-1833-4f0c-bd87-dd164f81f3af.png)

2. 3D disabled:

![image](https://user-images.githubusercontent.com/636095/189095545-d79492c0-d8ff-470c-8397-92af331395d0.png)

3. 3D disabled and slightly zoomed out:

![image](https://user-images.githubusercontent.com/636095/189095690-cbfc9771-d517-4ed7-9212-cdb3bea32412.png)

4. 3D enabled and slightly zoomed out:

![image](https://user-images.githubusercontent.com/636095/189095784-eb7e8c88-b114-43d6-852a-18626de06aa4.png)

**Comment by prozessor13** [comment]
Yep. in 2D tile-list generation anchor is map-center, in 3d camera-position. So in 2d you get more tiles to the horizon, but the error is the same.

**Comment by HarelM** [comment]
I'm not sure how to categorize this in terms of label - is this a bug or a enhancement? Is it only in 3D?
